lib/spack/spack/test/installer.py

```diff
@@ -252,6 +253,46 @@ def test_installer_str(install_mockery):
     assert "failed (0)" in istr
 
 
+def test_installer_prune_built_build_deps(install_mockery):
+    r"""
+    Ensure that build dependencies of installed deps are pruned
+    from installer package queues.
+
+                (a)
+               /   \
+             (d)   (b) <-- is installed already so we should
+              |      \     prune (c) from this install since
+             (e)     (c)   it is *only* needed to build (b)
```

Comment:
looks alright, but can you additionally test with a build edge between d and c, so that you're sure the installer is still visiting all edges, not just all nodes?

Reply:
I think I might be missing something here. If we add an additional build edge between (d) and (c), (c) will not be pruned from the DAG since (d) is not built. 

Perhaps a better way to do this would be to make (b) also depend on (e) and then ensure that (e) still remains in the DAG since it is also required by (d).

Reply:
I just mean that you should test that nodes are pruned if and only if *all* connecting edges are prunable build edges. The current test is insufficient as it would also pass when the implementation accidentally prunes nodes if *any* build edge to it is prunable.